scripts/ARCHITECTURE.md

```diff
@@ -0,0 +1,383 @@
+# Spyglass Scripts Architecture
+
+This document describes the execution flow, decision points, and design
+rationale for the installation and validation scripts.
+
+## Overview
+
+```
+┌─────────────────────────────────────────────────────────────────────┐
+│                        User Entry Points                            │
+├─────────────────────────────────────────────────────────────────────┤
+│  python scripts/install.py     Interactive or CLI installation      │
+│  python scripts/validate.py    Post-install health check            │
+└─────────────────────────────────────────────────────────────────────┘
+```
+
+## install.py - Main Installer
+
+### Execution Flow
+
+The installer follows a **strict execution order** to avoid circular
+dependencies (Spyglass cannot be imported until after it's installed).
+
+```
+┌──────────────────────────────────────────────────────────────────────┐
+│                         INSTALLATION FLOW                            │
+├──────────────────────────────────────────────────────────────────────┤
+│                                                                      │
+│  1. PARSE ARGUMENTS                                                  │
+│     └─→ CLI flags (--minimal, --docker, etc.)                        │
+│     └─→ Environment variables (SPYGLASS_BASE_DIR)                    │
+│                                                                      │
+│  2. DRY-RUN CHECK ─────────────────────────────────────────────────┐ │
+│     └─→ If --dry-run: show plan and exit                          │ │
+│                                                                    │ │
+│  3. DETERMINE INSTALLATION TYPE                                    │ │
+│     ├─→ --minimal flag → environment-min.yml                       │ │
+│     ├─→ --full flag → environment.yml                              │ │
+│     └─→ Neither → Interactive prompt                               │ │
+│                                                                    │ │
+│  4. GET BASE DIRECTORY                                             │ │
+│     ├─→ --base-dir CLI arg (highest priority)                      │ │
```

Comment:
It's not clear at this point what priority means. Only later did i realize it meant precedence over fallbacks